### PR TITLE
Closes #772: Migration for D7 plain text paragraph type.

### LINKED ISSUE #772: Migrate Plain Text paragraph type
See migration mapping doc for details: https://docs.google.com/spreadsheets/d/134UMiYtE8f1YAKYgSukL2JxFzDJF6-TgCAvYFC-vxqQ/edit#gid=0

### PR DESCRIPTION
**Pull request by smitahantzios**
Closes #772 
Migration for Plain Text Paragraph

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_plain_text.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+process:
+
+  field_az_text_area:
+    plugin: sub_process
+    source: field_uaqs_text_area
+    revision_id: revision_id
+    process:
+      delta: delta
+      value: value
+      summary: summary
```
Comment (tadean) [inline modules/custom/az_migration/migrations/migrate_plus.migration.az_paragraph_plain_text.yml:25]: `field_az_text_area` in the `az_text` paragraph does not have the summary subfield enabled so this probably should be omitted.

File: modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  public function prepareRow(Row $row) {
+
+    // Get the source paragraph item id.
+    $item_id = $row->getSourceProperty('item_id');
+
+    $paragraph_bottom_spacing = $this->select('paragraphs_item', 'pi')
+      ->fields('pi', ['bottom_spacing'])
+      ->condition('pi.item_id', $item_id)
+      ->execute()
+      ->fetchCol();
```
Comment (tadean) [inline modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php:43]: This runs a separate query for every paragraph it seems like outside of the migration query. Would it be better to add the new field in the overriden `ParagraphsItem::query()` function so this extra query isn't necessary?

Comment (smitahantzios) [reply modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php:43]: @tadean - Are you suggesting to add `AZParagraphsItem:query()` and then overwrite the `ParagraphsItem::query()` in the custom plugin ?

Comment (tadean) [reply modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php:43]: Yes, correct.

Comment (smitahantzios) [reply modules/custom/az_migration/src/Plugin/migrate/source/AZParagraphsItem.php]: Thanks. updated the code as per the discussion.

**Review by joeparsons — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**